### PR TITLE
Nits on SNI-DNAT: fixing helm, etc.

### PR DESCRIPTION
**Pull request by rshriram**
Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

### CODE REVIEW

File: install/kubernetes/helm/subcharts/gateways/templates/deployment.yaml [hunk 1]
```diff
@@ -145,9 +145,9 @@ spec:
           - name: ISTIO_META_POD_NAME
             valueFrom:
               fieldRef:
-                fieldPath: metadata.name                
-          {{- if $spec.envVariables }}
-          {{- range $key, $val := $spec.envVariables }}
```
Comment (rshriram) [inline install/kubernetes/helm/subcharts/gateways/templates/deployment.yaml]: because the field was renamed to env in helm per costin's suggestion

**Comment by douglas-reid** [comment]
we need this to unblock `release-1.1`, so:

/lgtm

File: pilot/pkg/networking/core/v1alpha3/cluster.go [hunk 2]
```diff
@@ -25,7 +25,6 @@ import (
 	v2_cluster "github.com/envoyproxy/go-control-plane/envoy/api/v2/cluster"
 	"github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
 	"github.com/gogo/protobuf/types"
-
```
Comment (douglas-reid) [inline pilot/pkg/networking/core/v1alpha3/cluster.go]: won't removing this line cause linter issues?  or did we recently stop having separation between outside libs and istio libs in the imports?

Comment (rshriram) [reply pilot/pkg/networking/core/v1alpha3/cluster.go]: oh go fmt or goimports did this, on goland IDE..

**Pull request merged**